test/tools/unified-spec-runner/match.ts

```diff
@@ -328,22 +408,31 @@ function compareEvents(
   }
   for (const [index, actualEvent] of actual.entries()) {
     const expectedEvent = expected[index];
+    const rootPrefix = `events[${index}]`;
 
     if (expectedEvent.commandStartedEvent) {
-      expect(actualEvent).to.be.instanceOf(CommandStartedEvent);
-      resultCheck(actualEvent, expectedEvent.commandStartedEvent, entities, [
-        `events[${index}].commandStartedEvent`
-      ]);
+      const path = `${rootPrefix}.commandStartedEvent`;
+      if (!(actualEvent instanceof CommandStartedEvent)) {
```

Comment:
the `instanceof` + `expect.fail` allows typescript to narrow the type, so that the call to `compareCommand<type>Events` knows it has an object of the correct type.